src/Http/Requests/SortAPIRequest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace EscolaLms\Courses\Http\Requests;
+
+use EscolaLms\Courses\Models\Topic;
+use EscolaLms\Courses\Models\Lesson;
+use EscolaLms\Courses\Models\Course;
+use Illuminate\Foundation\Http\FormRequest;
+
+class SortAPIRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        $user = auth()->user();
+        $course = Course::find($this->input('course_id'));
+
+        if (isset($user) && $user->can('update', $course)) {
+            $class = $this->input('class');
+            $ids = array_map(function ($order) {
+                return $order[0];
+            }, $this->input('orders'));
+
+            switch ($class) {
+                case "Lesson":
+                    $lessons = Lesson::whereIn('id', $ids)->get();
```

Comment:
Couldn't this query and loop be simplified to
```suggestion
           return Lesson::whereIn('id', $ids)->where('course_id', '!=', $course->id)->doesntExist();
```